### PR TITLE
Test OcdFileImport::fillPathCoords

### PR DESCRIPTION
**Pull request by dg0yt**
Extended version of the proposed test, https://github.com/OpenOrienteering/mapper/pull/2224#issuecomment-2258301038

### CODE REVIEW

File: test/file_format_t.cpp [hunk 1]
```diff
@@ -1306,6 +1311,234 @@ void FileFormatTest::ocdTextImportTest()
...
+
+enum OcdPathPersonality {
+	Line = 0,
+	Area = 1
+};
+Q_DECLARE_METATYPE(OcdPathPersonality)
+
+void FileFormatTest::ocdPathImportTest_data()
+{
+	#define C(x) ((int)((unsigned int)(x)<<8))  // FIXME: Not the same as in export
```
Comment (dg0yt) [inline test/file_format_t.cpp:1356]: As long as we don't check the actual x/y values, it doesn't need action.

File: test/file_format_t.cpp [hunk 2]
```diff
@@ -1306,6 +1311,234 @@ void FileFormatTest::ocdTextImportTest()
...
+	constexpr auto ocd_flag_gap = 8;  // TODO: implement as Ocd::OcdPoint32::FlagGap
```
Comment (dg0yt) [inline test/file_format_t.cpp:1357]: #2266

File: test/file_format_t.cpp [hunk 3]
```diff
@@ -1306,6 +1311,234 @@ void FileFormatTest::ocdTextImportTest()
...
+{
+	QFETCH(OcdPointsView, points);
+	QFETCH(OcdPathPersonality, personality);
+	QFETCH(FlagsView, expected);
+	
+	TestOcdFileImport ocd_v12_import{12};
+	
+	TestOcdFileImport::OcdImportedPathObject path_object;
+	ocd_v12_import.fillPathCoords(&path_object, personality == Area, points.size, points.data);
+	QVERIFY(path_object.getRawCoordinateVector().size() > 0);
```
Comment (dg0yt) [inline test/file_format_t.cpp:1525]: There was a time when `OcdPointsView` and `FlagsView` returned empty views, turning the test into a no-op.

File: test/file_format_t.cpp [hunk 4]
```diff
@@ -1306,6 +1311,234 @@ void FileFormatTest::ocdTextImportTest()
...
+	
+//	QEXPECT_FAIL("empty hole, area", "", Abort);
```
Comment (dg0yt) [inline test/file_format_t.cpp:1527]: Oops, left from WIP.

File: test/file_format_t.cpp [hunk 5]
```diff
@@ -1306,6 +1311,234 @@ void FileFormatTest::ocdTextImportTest()
...
+	QCOMPARE(path_object.getRawCoordinateVector().size(), expected.size);
+	
+	for (int i = 0; i < expected.size; ++i)
+	{
+		// QCOMPARE(path_object.getCoordinate(i).flags(), *expected);
+		// Provide the current index when failing.
+		if (path_object.getCoordinate(i).flags() != static_cast<MapCoord::Flags>(expected.data[i]))
+		{
+			auto err = QString::fromLatin1("Compared flags are not the same at index %1\n   Actual  : %2\n   Expected: %3")
+			.arg(QString::number(i), QString::number(path_object.getCoordinate(i).flags()), QString::number(expected.data[i]));
```
Comment (dg0yt) [inline test/file_format_t.cpp:1537]: Oops, wrong indent.

**Comment by dg0yt** [comment]
Ping @dl3sdo. This has a more direct impact on what can be merged next.

**Comment by dl3sdo** [comment]
@dg0yt: Just to understand it right: this unit test is intentionally designed for the 'old' behaviour, i.e. merging #2224 would require an update of this unit test.
Because the expected behaviour of, e.g.,
```
MapCoord::ClosePoint | MapCoord::HolePoint,
MapCoord::ClosePoint | MapCoord::HolePoint,
MapCoord::ClosePoint | MapCoord::HolePoint,
```
passes and should not pass after #2224.
I assume that your approach shall allow to track the effects of #2224.

**Comment by dl3sdo** [comment]
Kai: this unit test is excellent: it's understandable, easily extensible and it documents the mapping of .ocd coords and attributes to Mapper's coords and attributes. Thank you for the effort you spent on it!

I added another test case but cannot push to your branch, please pick it from
https://github.com/OpenOrienteering/mapper/commit/f6d6682cb76e16c7c5c8731580d6bb532dd92dda
if you consider it to be useful.

**Comment by dg0yt** [comment]
> this unit test is intentionally designed for the 'old' behaviour, i.e. merging #2224 would require an update of this unit test.

Yes, for the moment. It is sort-of a documentation of what is implemented now.
Another commit can add the changes in output that we want to see.
Then another change do the actual modifications to the tested feature.

**Pull request merged**